sympy/printing/pretty/tests/test_pretty.py

```diff
@@ -7824,3 +7827,34 @@ def test_deprecated_prettyForm():
 
     with warns_deprecated_sympy():
         assert p.unicode == p.s == 's'
+
+
+def test_print_mat_add():
+    # Issue 23552
+    A = MatrixSymbol('A', 1, 1)
+    b = Symbol('b')
+    assert pretty(MatAdd(evaluate=False)) == '0'
+    assert pprint_out(MatAdd(evaluate=False)) == '𝟘'
+    assert pretty(MatAdd(A, evaluate=False)) == 'A'
+    assert pretty(MatAdd(b, evaluate=False)) == 'b'
+    assert pretty(MatAdd(1, evaluate=False)) == '1'
+    assert pretty(MatAdd(1, b, evaluate=False)) == '1 + b'
+    assert pretty(MatAdd(b, 1, evaluate=False)) == 'b + 1'
```

Comment:
These things should not be in the test suite. This is an invalid use of `MatAdd` and it should be disallowed. This test deliberately asserts that `MatAdd` can be used in ways that break its internal invariants. This kind of thing just makes it hard in future to enforce or depend on those invariants in the codebase (at which point someone like me will come along and strip out all these tests as being invalid).

Whether or not this should raise an exception or emit a warning is a question that can reasonably be asked. We shouldn't add regression tests for invalid usage though.

Rather than trying to make this work can we please focus on how to better serve the usecase in a way that can reasonably be supported:

1. Maybe there could be a `PrintAdd` class whose sole purpose is for printing
2. Maybe we should make it easier somehow to use `MatrixSymbol`.
3. Something else?

I don't see why so much effort is being expended here on making it possible to do something in a bad way when there is already a perfectly good way: use MatrixSymbol.

Reply:
If I might say - I think the effort being expended is just because it's still not clear what the right thing to do is.

I see that these should in some sense not be allowed, but it is in fact possible to create these objects, and so you get a very obscure error from `pprint`.

I really have no skin in this game - I can solve my problem another way.    It just seemed to me that the current error in `pprint` or `pretty` was probably not the best possible outcome - and that the problem needed to be fixed elsewhere - possibly to prevent these objects being created.  But seemed out of scope.

I just put the tests in because I wanted to test I'd fixed the undesirable behavior.  Why not just leave them in, then remove the ability to create these objects, and then remove the tests.   But - as I say - totally up to y'all.

Reply:
I think the proper fix is for MatAdd to error when given non matrices as arguments. It used to. I don't know why that was changed. I'm not sure (given that at least one person is apparently using this) whether there should be a deprecation for this even though it was never something that was "supported".

Reply:
That's totally fine by me - as I say.   This current fix is only for the case where you haven't done that yet...